surface/model_openapiv3.go

```diff
@@ -185,40 +233,75 @@ func (b *OpenAPI3Builder) buildTypeFromParameters(
 			}
 			f.Serialize = true
 			t.addField(&f)
+		} else if parameterRef := parametersItem.GetReference(); parameterRef != nil {
+			f.Type = typeForRef(parameterRef.GetXRef())
+			f.Name = strings.ToLower(f.Type)
+			f.Serialize = false
+			f.Kind = FieldKind_REFERENCE
+			t.addField(&f)
 		}
 	}
+
+	_, err = b.buildTypeFromRequestBody(name, requestBody, t)
+
+	if len(t.Fields) > 0 {
+		b.model.addType(t)
+		return t.Name, err
+	}
+	return "", err
+}
+
+// buildTypeFromRequestBody builds a service type description from the request bodies of an OpenAPI
+// description. If tIn is not given, a new type is created. Otherwise tIn is used.
+func (b *OpenAPI3Builder) buildTypeFromRequestBody(name string, requestBody *openapiv3.RequestBodyOrReference, tIn *Type) (tOut *Type, err error) {
+	if tIn == nil {
+		tOut = &Type{}
+		tOut.Name = name
+	} else {
```

Comment:
Personally, I don't like to use `else` statements if I can. They make the code a little harder to read, having to jump back up and through the `if`/`else if` statements to build the context in which the `else` body is being run.

I would do default initialization first, then do the check `tIn != nil` case. Like:

```go
tOut = &Type{
  Name: name,
}

if tIn != nil {
  tOut = tIn
}
```

This is easier to read: create a new Type with `name`, but if given one, use that instead.

Reply:
Done